src/gui/src/gui.cpp

```diff
@@ -77,6 +78,9 @@ static odb::dbBlock* getBlock(odb::dbDatabase* db)
 // This provides the link for Gui::redraw to the widget
 static gui::MainWindow* main_window = nullptr;
 
+// Placement density heatmap
+static std::unique_ptr<PlacementDensityDataSource> placement_density_heat_map = nullptr;
```

Comment:
Is there a reason this isn't a class member?

Reply:
overlooked, I moved it to the Gui class.